wagtail/api/v2/filters.py

```diff
@@ -231,4 +231,7 @@ def filter_queryset(self, request, queryset, view):
             for hook in hooks.get_hooks('construct_explorer_page_queryset'):
                 queryset = hook(parent_page, queryset, request)
 
+            user_perms = UserPagePermissionsProxy(request.user)
+            queryset = queryset & user_perms.explorable_pages()
```

Comment:
I'm torn over whether this filter should be included as part of the `for_explorer` flag, or be a separate one such as `explorable=1`. It slightly undermines the point of an API if the consumers of it have to identify themselves as "I am the explorer" and have their responses tuned accordingly, as opposed to the API parameters explicitly describing what the API should do. `for_explorer` is a necessary evil in this respect, because the `construct_explorer_page_queryset` hook already existed and logically has to be triggered on any bit of UI that self-identifies as "the explorer"... I'd prefer not to go further down this path if we can help it.

On a related note, I'm wary that this is the first time we've introduced an API feature that depends on the user's credentials, via `request.user`. In the context of the admin API this makes sense, because we know that the HTTP request will be coming from the browser session where the user is logged in - but for a general-purpose API that could be an unsafe assumption, and we might want to look at a more generalised authentication system that can support things like tokens and OAuth ([django-rest-framework](https://www.django-rest-framework.org/api-guide/authentication/) can hopefully do a lot of the heavy lifting here). Or, if we don't want to go down that path just yet, it might make sense to move it into wagtail/admin/api/filters.py to make it explicitly an admin-API-only feature.

(@kaedroho, any thoughts on the above?)

Reply:
Having read up on https://www.django-rest-framework.org/api-guide/authentication/ , I'm now satisfied that the use of `request.user` isn't a problem - DRF's pluggable authentication backends populate `request.user` as standard, regardless of the authentication mechanism.